### PR TITLE
Add inplace param to flatten operation. 

### PR DESCRIPTION
**Pull request by ducleminh**
### Description and related tickets, documents
Add inplace param to flatten operation, moved flatten (array type) unit tests to DDF core.
- https://adatao.atlassian.net/browse/PE-1902

Reviewers: @hai-adatao @Huandao0812 @nhanitvn @phvu 

### Breaking changes & backward compatible issues
Is this PR a breaking change or has backward compatible issue (e.g: changes in API names, interfaces, signature / remove something...)?
If yes, please state out what and tag, and if it can possibly affect BA, please tag aht, tuananh, khangpham and baonguyen

### How to test
Describe how this PR is tested. In case manual testing is required, describe how to do so.

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [ ] Merge check has no conflicts. PR checks passed.
- [ ] Code review is done.

### CODE REVIEW

**Comment by hai-adatao** [comment]
What is the difference between `flatten` and `flattenArrayColumn` @binhmop @ducleminh ? I guess one for object type and one for array type, right?
If that's the case, can you help put in some tests for `flatten` @ducleminh ?

**Comment by ducleminh** [comment]
yeah, flattenDDF applies to structtype whereas the other, as the name suggest will flatten array-type column.
unit tests for flatten added. @hai-adatao

File: ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala [hunk 1]
```diff
@@ -238,4 +238,58 @@ trait TransformationHandlerBaseSuite extends BaseSuite with Matchers {
...
+    elementStr should be("month")
+  }
+
+  test("test flatten column of array type inPlace False") {
+    val inPlace = false
+    val ddf: DDF = loadAirlineDDF().VIEWS.project("Year", "Month", "DayofMonth", "DayofWeek", "DepTime", "CRSDepTime", "ArrTime", "CRSArrTime")
+    val ddfWithArrayTypeColumn: DDF = ddf.Transform.transformUDF("arrCol = array(1.234, 5.678, 9.123)")
+    val flattenedDDF: DDF = ddfWithArrayTypeColumn.Transform.flattenArrayTypeColumn("arrCol", inPlace)
+
+    ddfWithArrayTypeColumn should not be null
```
Comment (hai-adatao) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: Dup

File: ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala [hunk 2]
```diff
@@ -238,4 +238,58 @@ trait TransformationHandlerBaseSuite extends BaseSuite with Matchers {
...
+    flattenedDDF.getColumnName(11) should be("arrCol_c2")
+  }
+
+  test("test flatten column of array type inPlace True") {
+    val inPlace = true
+    val ddf: DDF = loadAirlineDDF().VIEWS.project("Year", "Month", "DayofMonth", "DayofWeek", "DepTime", "CRSDepTime", "ArrTime", "CRSArrTime")
+    val ddfWithArrayTypeColumn: DDF = ddf.Transform.transformUDF("arrCol = array(1.234, 5.678, 9.123)")
+    val flattenedDDF: DDF = ddfWithArrayTypeColumn.Transform.flattenArrayTypeColumn("arrCol", inPlace)
+
+    ddfWithArrayTypeColumn should not be null
```
Comment (hai-adatao) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: Dup

**Comment by hai-adatao** [comment]
It's not in the scope of this ticket, but I want to discuss 2 things, and possibly we can create tickets to track it too:
- Can we move implementation of `flatten` to core instead of spark?
- Add tests for flatten array of struct and struct of array, if we can cover these 2 then we can be confident that `flatten` and `flattenArrayTypeColumn` works most of the cases

@nhanitvn @Huandao0812

**Pull request merged**